### PR TITLE
Updating proxy doc for clarity and grammar

### PR DESCRIPTION
**Pull request by coopr**
I plan to propose more changes... have to get on a call now... feel free to teach me how to do this better ;)

### CODE REVIEW

File: app/docs/0.10.x/proxy.md [hunk 1]
```diff
@@ -138,15 +125,13 @@ HTTP/1.1 201 Created
...
+"http://my-api.com". It also specifies various routing properties, though note that **only one of** `hosts`, `uris` and `methods` is  required. 
 
 Adding such an API would mean that you configured Kong to proxy all incoming 
-requests matching `hosts`, `uris`, and `methods` to `http://my-api.com`.
-Kong being a transparent proxy, it will forward the request to your upstream
-service - practically - untouched (exception made of various headers such as
-`Connection`, or in case a plugin performs some transformations to it).
+requests matching the specified `hosts`, `uris`, and `methods` to `http://my-api.com`.
+Kong is a transparent proxy, and it will forward the request to your upstream
+service untouched, with the exception of the addition of various headers such as
```
Comment (nijikokun) [inline app/docs/0.10.x/proxy.md]: can remove `of the addition` as now we repeat `of` twice in a single sentence in close proximity and we reduce sentence complexity for brevity

File: app/docs/0.10.x/proxy.md [hunk 2]
```diff
@@ -94,18 +87,15 @@ where requests are forwarded to.
...
+request upstream to your own API or service. 
+
+When a client makes a request to the proxy port, Kong will decide to which upstream service or API to route
+(or forward) the incoming request - depending on the API configuration in Kong, which is managed via the Admin 
+API. You can configure APIs with various properties, but the three relevant 
 ones for routing incoming traffic are `hosts`, `uris` and `methods`.
 
-If Kong cannot determine to which API a given request should be routed to, it
-will response with:
+If Kong cannot determine to which upstream API a given request should be routed, Kong
```
Comment (nijikokun) [inline app/docs/0.10.x/proxy.md]: `upstream` seems too verbose here

File: app/docs/0.10.x/proxy.md [hunk 3]
```diff
@@ -74,18 +71,14 @@ It is worth pointing out that both of these ports can be configured. See the
 
 ### Terminology
 
-In this document, we will refer to various Kong concepts. For the sake of 
-consistency, we will now define terms refering to such concepts:
-
 - `API`: This term refers to the API entity of Kong. You configure your APIs,
-that point to your own Upstream services through the Admin API.
+that point to your own Upstream services, through the Admin API.
```
Comment (p0pr0ck5) [inline app/docs/0.10.x/proxy.md]: upstream should be lowercase

File: app/docs/0.10.x/proxy.md [hunk 4]
```diff
@@ -74,18 +71,14 @@ It is worth pointing out that both of these ports can be configured. See the
...
 - `Plugin`: This refers to Kong "plugins", which are pieces of business logic
-that run in proxying lifecycle. Plugins can be configured globally (all 
-incoming traffic), or on a per-API basis through the Admin API.
-- `Client`: Refers to the - downstream - client making requests to Kong's proxy
+that run in the proxying lifecycle. Plugins can be configured through the Admin API - either globally (all incoming traffic) or on a per-API basis.
```
Comment (p0pr0ck5) [inline app/docs/0.10.x/proxy.md]: replace the dash with a comma

File: app/docs/0.10.x/proxy.md [hunk 5]
```diff
@@ -74,18 +71,14 @@ It is worth pointing out that both of these ports can be configured. See the
...
+- `Client` or : Refers to the *downstream* client making requests to Kong's proxy
```
Comment (p0pr0ck5) [inline app/docs/0.10.x/proxy.md]: "client or"? or should be removed?

File: app/docs/0.10.x/proxy.md [hunk 6]
```diff
@@ -94,18 +87,15 @@ where requests are forwarded to.
...
-relevant to this guide.
-
-When a client makes a request to the proxy port, Kong will decide where to route
-(or forward) that incoming request (that is, to which of your upstream 
-services). Kong achieves this through the APIs you configured (via the Admin 
-API). You can configure APIs with various properties, but the three relevant 
+request upstream to your own API or service. 
+
+When a client makes a request to the proxy port, Kong will decide to which upstream service or API to route
+(or forward) the incoming request - depending on the API configuration in Kong, which is managed via the Admin
```
Comment (p0pr0ck5) [inline app/docs/0.10.x/proxy.md]: again, comma instead of dash. its not a hard stop, just a sentance fragment

File: app/docs/0.10.x/proxy.md [hunk 7]
```diff
@@ -138,31 +125,25 @@ HTTP/1.1 201 Created
...
+service untouched, with the exception of the addition of various headers such as
+`Connection`.
 
 [Back to TOC](#table-of-contents)
 
 ### Routing capabilities
 
-Let's now investigate in details how Kong matches a request to the configured
-`hosts`, `uris` and `methods` properties of your API:
+Let's now understand how Kong matches a request to the configured
```
Comment (p0pr0ck5) [inline app/docs/0.10.x/proxy.md]: s/understand/discuss/?

File: app/docs/0.10.x/proxy.md [hunk 8]
```diff
@@ -412,33 +399,33 @@ property for that:
...
-Will make Kong send the following request to your upstream service:
+Will cause Kong to send the following request to your upstream service:
 
 ```http
 GET /path/to/resource HTTP/1.1
 Host: my-api.com
 ```
 
-This property can come very handy at times.
+This property can be very handy at times.
```
Comment (p0pr0ck5) [inline app/docs/0.10.x/proxy.md]: at what times? this sentence is probably not necessary :)

File: app/docs/0.10.x/proxy.md [hunk 9]
```diff
@@ -412,33 +399,33 @@ property for that:
...
 
 [Back to TOC](#table-of-contents)
 
 #### Request HTTP method
 
-Starting with Kong 0.10, client requests can also be routed depending on their
+Starting with Kong `0.10`, client requests can also be routed depending on their
```
Comment (p0pr0ck5) [inline app/docs/0.10.x/proxy.md]: covering the version number in ticks feels unnecessary

File: app/docs/0.10.x/proxy.md [hunk 10]
```diff
@@ -460,25 +447,25 @@ HEAD /resource HTTP/1.1
...
+the authentication and rate limiting plugins to such requests).
 
 [Back to TOC](#table-of-contents)
 
 ### Routing priorities
 
-It is now clear that an API may define matching rules based on its `hosts`, 
-`uris`, and `methods` fields. It is also clear that for Kong to match an
+An API may define matching rules based on its `hosts`, 
+`uris`, and `methods` fields. For Kong to match an
```
Comment (p0pr0ck5) [inline app/docs/0.10.x/proxy.md]: please ensure consistent use of oxford comma (line 95 is an example of alternative, pick and stick with one :) )

File: app/docs/0.10.x/proxy.md [hunk 11]
```diff
@@ -735,7 +716,7 @@ standard HTTP proxy.
 ### Conclusion
 
 Through this guide, we hope you gained knowledge of the underlying proxying
-mehanism of Kong, from how is a request matched to an API, to how to allow for
+mechanism of Kong, from how is a request matched to an API, to how to allow for
 using the WebSocket protocol or setup SSL for an API.
 
 This website is Open-Source and can be found at
```
Comment (p0pr0ck5) [inline app/docs/0.10.x/proxy.md]: open source, lower case, no dash

File: app/docs/0.10.x/proxy.md [hunk 12]
```diff
@@ -590,39 +571,39 @@ the client.
...
 One exception to this is made when using the WebSocket protocol. If so, Kong
-will set the following headers to allow for upgrading the protocol bewteen the
+will set the following headers to allow for upgrading the protocol between the
 client and your upstream services:
 
 - `Connection: Upgrade`
 - `Upgrade: websocket`
 
-More information on this topic will be covered in the
+More information on this topic in covered in the
```
Comment (p0pr0ck5) [inline app/docs/0.10.x/proxy.md]: "on this topic **is** covered"

**Comment by p0pr0ck5** [comment]
@coopr left some comments. Also, let's try to format this so that lines are no longer than 80 characters. Please feel free to poke me if you need a hand :)

File: app/docs/0.10.x/proxy.md [hunk 13]
```diff
@@ -4,29 +4,26 @@ title: Proxy Reference
...
 similar behavior as the `:8000` port, except that it expects HTTPs traffic. This
 port can be disabled via the configuration file.
 
-We will cover routing capabilities of Kong by explaining in details how incoming 
+In this document we cover routing capabilities of Kong by explaining in detail how incoming 
 requests on port `:8000` are proxied to a configured upstream service depending
 on their headers, URI, and HTTP method.
 
-It is worth pointing out that both of these ports can be configured. See the
-[Configuration Reference] for more details.
```
Comment (thibaultcha) [inline app/docs/0.10.x/proxy.md]: Why removing this?

Comment (coopr) [inline app/docs/0.10.x/proxy.md]: I guessed that most readers would assume ports are configurable, and that most readers wouldn't want to configure ports at this point of reading the docs, and if they wanted to find out how to configure ports, they could easily do so by searching the docs. I don't have a strong preference here.

**Pull request merged**